File: src/src/main/java/com/mewebstudio/translatable/entity/AbstractBaseEntity.java
package com.mewebstudio.translatable.entity;

import com.mewebstudio.translatable.entity.generator.GeneratedId;
import jakarta.persistence.Column;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.io.Serializable;
import java.time.LocalDateTime;

@Setter
@Getter
@NoArgsConstructor
@SuperBuilder
@MappedSuperclass
public abstract class AbstractBaseEntity implements Serializable {
    @Id
    @GeneratedId
    @Column(name = "id", nullable = false, updatable = false, length = 26)
    private String id;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: src/main/kotlin/com/mewebstudio/translatable/entity/AbstractBaseEntity.kt
package com.mewebstudio.translatable.entity

import com.mewebstudio.translatable.entity.generator.GeneratedId
import jakarta.persistence.Column
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.io.Serializable
import java.time.LocalDateTime

@MappedSuperclass
abstract class AbstractBaseEntity : Serializable {
    @Id
    @GeneratedId
    @Column(name = "id", nullable = false, updatable = false, length = 26)
    lateinit var id: String

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    lateinit var createdAt: LocalDateTime

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    lateinit var updatedAt: LocalDateTime
}
